Task3/Exceptions/DuplicateUserException.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace Task3.Exceptions
+{
+    [Serializable]
```

Comment:
https://stackoverflow.com/questions/1066701/why-should-i-always-make-my-exceptions-serializable-net

Reply:
As I understand you should Apply this only in case you are not handling exception by your code and want to keep as much info as possible. So you should understand is it really necessary to support Serializable. Not always it s good to share error context with other applications, better to handle it by try catch block and log it. 

https://docs.microsoft.com/en-us/dotnet/api/system.exception?view=net-6.0#Custom